server/lemmy_db/src/user.rs

```diff
@@ -125,9 +126,18 @@ impl User_ {
     use crate::schema::user_::dsl::*;
     user_.filter(actor_id.eq(object_id)).first::<Self>(conn)
   }
-}
 
-impl User_ {
+  pub fn find_by_email_or_username(
+    conn: &PgConnection,
+    username_or_email: &str,
+  ) -> Result<Self, Error> {
+    if is_email_regex(username_or_email) {
+      Self::find_by_email(conn, username_or_email)
+    } else {
+      Self::find_by_username(conn, username_or_email)
+    }
+  }
```

Comment:
Is there any reason why you moved this back into the database? I think its fine to have it as a higher level function thats provided by the API itself.

Reply:
Yeah, there was a TODO that it should probably be moved into the DB, plus all the other `read_from`s and find_bys are there.